src/components/Tab.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import * as Conseil from '../Conseil';
+import { ConseilFilter } from '../Conseil';
+import { DataView } from '../types';
+import { ExpandableGrid } from './ExpandableGrid';
+
+interface TabProps {
+    dataView: DataView;
+    hidden: boolean;
+    filters: ConseilFilter;
+    network: string;
+}
+
+interface TabState {
+    data: Object[];
+}
+
+export class Tab extends React.Component<TabProps, TabState> {
+
+    constructor(props: TabProps) {
+        super(props);
+        this.state = {data: []};
+        this.refreshData = this.refreshData.bind(this);
+        this.refreshData(props);
+    }
+
+    refreshData(nextProps: TabProps) {
+        if (!nextProps.hidden) {
+            let funcToCall = Conseil.getBlocks;
+            if (this.props.dataView === DataView.Operations) {funcToCall = Conseil.getOperations; }
+            if (this.props.dataView === DataView.Accounts) {funcToCall = Conseil.getAccounts; }
+            funcToCall.apply(null, [this.props.network, nextProps.filters]).
```

Comment:
Rather than `.apply` to spread arguments, you can use the new es6 spread operator:
`funcToCall(...[this.props.network, nextProps.filters])`

Reply:
Doesn't look like it works with TypeScript yet. :disappointed: 

https://github.com/Microsoft/TypeScript/issues/18183